server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java

```diff
@@ -108,10 +108,10 @@ public MonitorScheduler getMonitorScheduler(
 
     return new MonitorScheduler(
         config.get(),
-        CronScheduler.newBuilder(Duration.ofSeconds(1L)).setThreadName("MonitorScheduler-%s").build(),
+        CronScheduler.newBuilder(Duration.ofSeconds(1L)).setThreadName("MonitorSchedulerThread").build(),
         emitter,
         monitors,
-        Executors.newCachedThreadPool()
+        Execs.multiThreaded(64, "MonitorThread-%d")
```

Comment:
This looks like an overkill. What was the rationale for using this many threads?

Reply:
Hi @jihoonson , sorry for delay in response. 
I think currently there are ~20 monitors, which can run concurrently with the MonitorScheduler class. Suppose a case in which frequency of scheduling < time taken by the executor thread to do `monitor.monitor(...)`(Although I am not sure if this case is possible in practical, kind of edge case). This can result in queuing of the tasks if threads are very less. I think we should atleast have no. of threads equal to max number of monitors supported. I may be missing something here. What do you think?

Reply:
The monitor usually takes less than 1 sec (probably less than 100 ms) while the emission period is large enough to run all monitors (1 min by default). So, I think the scenario you described can happen when there are some failures such as retrying metrics emission due to some network issue. However, I don't think we should handle these failures by employing multiple threads because there is nothing we can do better with more threads. I would rather not schedule a new monitor task if the previous one is still running. I implemented this in #10732.

Reply:
This makes sense, we should reduce the number of monitor threads.